### PR TITLE
fix(esbuild): only pass valid esbuild transform options

### PR DESCRIPTION
**Pull request by zoeyzhao19**
For now,  invalid `rollup.esbuild` options that aren't supported by `esbuild transform option` may breaks build, as #256 has shown.
This **PR** would fix:
- any esbuild option passed to unbuild should not break esbuild transform process
- since `loader` passed to `rollup.esbuild` have been overrided by inline handle logic, so inlined **esbuild option declaration** is also updated to reduce confusion. 

There is still one point as mentioned by #256, `tsconfig` has no effects in inline esbuild transform. I don't know would unbuild like to use it to resolve tsconfig file like `rollup-plugin-esbuild`, so I remained it unhandled.

### CODE REVIEW

File: src/builder/plugins/esbuild.ts [hunk 1]
```diff
@@ -78,10 +85,10 @@ export function esbuild(options: Options): Plugin {
       }
 
       const result = await transform(code, {
-        ...options,
+        ...esbuildOptions,
         loader,
         sourcefile: id,
-        sourcemap: options.sourceMap ?? options.sourceMap,
+        sourcemap: sourceMap,
```
Comment (ntnyq) [inline src/builder/plugins/esbuild.ts:91]: sourcemap: esbuildOptions.sourcemap ?? sourceMap,

Comment (ntnyq) [inline src/builder/plugins/esbuild.ts:91]:
```suggestion
        sourcemap: esbuildOptions.sourcemap ?? sourceMap,
```

Comment (zoeyzhao19) [reply src/builder/plugins/esbuild.ts:91]: @ntnyq  maybe `sourceMap ?? esbuildOptions.sourcemap`  to keep `sourceMap `  prior , and be consistent with the current

**Comment by pi0** [comment]
Hi and thank you for this PR ❤️  Sorry it took long to review. I have made a rework in #278

**Pull request closed without merging**